inc/Editor/BlockManagement/ConfigRegistry.php

```diff
@@ -48,84 +92,102 @@ public static function register_block( array $block_config = [] ): bool|WP_Error
+				// Validate the required query mapping.
+				$validation_result = self::validate_query_mapping( $input_schema, $required_query_input_schema, $required_query_output_schema, $block_title, $required_query_key, $required_query_type );
+				if ( is_wp_error( $validation_result ) ) {
+					return $validation_result;
 				}
-			}
 
-			if ( self::SEARCH_QUERY_KEY === $from_query_type ) {
-				$search_input_count = count( array_filter( $from_input_schema, function ( array $input_var ): bool {
-					return 'ui:search_input' === $input_var['type'];
-				} ) );
+				// Add the selector for the required query, noting that the input schema is the display query's input schema.
+				$selectors[] = [
+					'display_name' => self::get_query_name_from_key( $required_query_key ),
+					'image_url' => $required_query->get_image_url(),
+					'inputs' => self::map_input_variables( $input_schema ),
+					'name' => ucfirst( $required_query_key ),
+					'query_key' => $required_query_key,
+					'type' => $required_query_type,
+					'query_group' => $query_key,
+				];
 
-				if ( 1 !== $search_input_count ) {
-					return self::create_error( $block_title, 'A search query must have one input variable with type "ui:search_input"' );
-				}
+				// Add the required query to the queries array, so it won't be processed again.
+				$queries[ $required_query_key ] = $required_query;
 			}
 
-			// Add the selector to the configuration.
-			array_unshift(
-				$config['selectors'],
-				[
-					'image_url' => $from_query->get_image_url(),
-					'inputs' => self::map_input_variables( $input_schema ),
-					'name' => $selection_query['display_name'] ?? ucfirst( $from_query_type ),
-					'query_key' => $from_query::class,
-					'type' => $from_query_type,
-				]
+			// The query is either a display, or a list query.
+			$is_collection = true === ( $output_schema['is_collection'] ?? false );
+			$has_required_variables = array_reduce(
+				array_column( $input_schema, 'required' ),
+				fn( $carry, $required ) => $carry || ( $required ?? true ),
+				false
 			);
+
+			// Generate the selector for the query.
+			$selectors[] = [
+				'display_name' => self::get_query_name_from_key( $query_key ),
+				'image_url' => $query->get_image_url(),
+				'inputs' => self::map_input_variables( $input_schema ),
+				'name' => self::DISPLAY_QUERY_KEY === $query->get_type() ? ( $has_required_variables ? 'Manual input' : ( $is_collection ? 'Load collection' : 'Load item' ) ) : ucfirst( $query_key ),
+				'query_key' => $query_key,
+				'type' => $has_required_variables ? 'manual-input' : 'load-without-input',
+				'query_group' => $query_key,
+			];
+
+			// If the query is a display query, set the display query found flag.
+			if ( self::DISPLAY_QUERY_KEY === $query->get_type() ) {
```

Comment:
This feels like a core disconnect for me. _Any_ query provided to the block config can be a display query. The user chooses how to curate data; after having done so, we can look at the output schema and decide which queries are compatible as a display query. Very likely it's just one, but maybe the user should choose again.

Reply:
The core problem is:

How do we decide what query is a display query, and what query is a selector query? 

I tried to answer by having the user mark that down in the query config. But, while trying to fix the block bindings I realized the better question to ask is:

Is the query being executed a display query or a selector query?

This question is better as it implies, it can be dynamically adjusted. Is the search query driving the get art query? Great, it's a selector query. Is the search query not driving any other query but just rendering some results in a block? Great, it's a display query. That's roughly what you are getting at if I'm understanding you and I do agree with it.

Reply:
From the internal discussion, this would be as simple as them declaring this as

`"display_queries" => [ "get_art" ]`

This does make things a bit more simpler, though we aren't inferring whether a query is a selector or a display query based on what's declared.